presto-main/src/main/java/com/facebook/presto/operator/scalar/ReflectionParametricScalar.java

```diff
@@ -144,7 +146,13 @@ public ScalarFunctionImplementation specialize(BoundVariables boundVariables, in
         Signature boundSignature = SignatureBinder.bindVariables(getSignature(), boundVariables, arity);
         if (exactImplementations.containsKey(boundSignature)) {
             Implementation implementation = exactImplementations.get(boundSignature);
-            return new ScalarFunctionImplementation(implementation.isNullable(), implementation.getNullableArguments(), implementation.getMethodHandle(), isDeterministic());
+            MethodHandleAndConstructor methodHandleAndConstructor = implementation.specialize(boundSignature, boundVariables, typeManager, functionRegistry);
+            if (methodHandleAndConstructor != null) {
+                return new ScalarFunctionImplementation(implementation.isNullable(), implementation.getNullableArguments(), methodHandleAndConstructor.getMethodHandle(), methodHandleAndConstructor.getConstructor(), isDeterministic());
+            }
+            else {
+                return new ScalarFunctionImplementation(implementation.isNullable(), implementation.getNullableArguments(), implementation.getMethodHandle(), isDeterministic());
```

Comment:
Why would `methodHandleAndConstructor` be `null`? It doesn't seem like that should happen for an "exact" implementation